### DISCUSSION TITLE
Set a mask with Cyrillic characters in InputMask

### DISCUSSION THREAD

**Discussion by Thr0TT1e** [primefaces/community#1525] [PrimeVue]
Greetings.

Could you please advise how to enable Cyrillic in the InputMask component, if the mask attribute accepts the string.

<InputMask
       id="decimalNumber"
       v-model="decimal"
       mask="ЯЯЯЯ.99999-99"
       placeholder="ПППП.ХХХХХ-ХХ"
/>

I need the pattern to be `[А-Я]{4}.99999-99''.

Stack: Vue 3, PrimeVue.

**Comment by magicoff** [comment]
I can confirm this issue. The `InputMask` component doesn't support Cyrillic (or any non-Latin) characters because the built-in definitions only accept Latin letters:

```javascript
// InputMask.vue — line ~370
this.defs = {
    9: '[0-9]',
    a: '[A-Za-z]',
    '*': '[A-Za-z0-9]'
};
```

The `a` symbol uses `[A-Za-z]` regex, which excludes Cyrillic, Arabic, Chinese, and all other non-Latin scripts.

### Workaround

I created a custom component based on the original `InputMask` source with extended definitions:

```javascript
// CyrillicInputMask.vue
this.defs = {
    9: '[0-9]',
    a: '[A-Za-z]',           // Latin only
    ы: '[А-Яа-яЁё]',         // Cyrillic only (custom symbol)
    '*': '[A-Za-zА-Яа-яЁё0-9]'  // Latin + Cyrillic + digits
};
```

Usage example:

```vue

<CyrillicInputMask 
    v-model="ingotNumber" 
    mask="ы{9}.99999-99" 
    placeholder="ы{_}._____-__" 
    slotChar="_" 
/>

```

### Suggestion

It would be great if the core `InputMask` could support Unicode letters out of the box, perhaps by using `\p{L}` (Unicode property escape) instead of `[A-Za-z]`:

```javascript
// Proposed improvement
this.defs = {
    9: '[0-9]',
    a: '\\p{L}',              // Any Unicode letter (Latin, Cyrillic, Arabic, etc.)
    '*': '[\\p{L}0-9]'        // Any Unicode letter + digits
};
```

Note: `\p{L}` requires the `u` flag on the RegExp: `new RegExp('\\p{L}', 'u')`.

This would make InputMask truly international without needing custom symbols for each language.

**Reply by magicoff** [reply]
use 'ы' instead ы
